src/components/views/spaces/SpaceTreeLevel.tsx

```diff
@@ -29,39 +29,73 @@ interface IItemProps {
     isNested?: boolean;
 }
 
-const SpaceItem: React.FC<IItemProps> = ({
-    space,
-    activeSpaces,
-    isNested,
-}) => {
-    const isActive = activeSpaces.includes(space);
-    const classes = classNames("mx_SpaceButton", {
-        mx_SpaceButton_active: isActive,
-    });
-    const notificationState = SpaceStore.instance.getNotificationState(space.roomId);
-    const childSpaces = SpaceStore.instance.getChildSpaces(space.roomId);
-    const childItems = childSpaces ? <SpaceTreeLevel
-        spaces={childSpaces}
-        activeSpaces={activeSpaces}
-        isNested={true}
-    /> : null;
-    let notifBadge;
-    if (notificationState) {
-        notifBadge = <NotificationBadge forceCount={false} notification={notificationState} />;
+interface IItemState {
+    collapsed: boolean;
+}
+
+export class SpaceItem extends React.PureComponent<IItemProps, IItemState> {
```

Comment:
It might be helpful to keep this written in Functional using hooks to aid with accessibility work going forward where the roving tab index stuff works easiest.

Reply:
I can, but I thought we said to not use more hooks as some people on the team, including myself back then, found them confusing? Maybe I'm misremembering or that decision has reverted since?